### DISCUSSION TITLE
How hard would it be to expose the tool as a standard library to consume by other applications ?

### DISCUSSION THREAD

**Discussion by Greybird** [tomchavakis/nuget-license#124] [Ideas]
Hi,

I'm using the tool to scan a lot of code repositories for my organization for license issues.
I've built a dotnet executable, which in turns runs the dotnet-project-licenses tool, with a json output, and then, consumes this json file.

I was wondering how hard it would be, on top of the tool exposition, to build a separate (or not ?) nuget, exposing an api to retrieve the information about licenses directly, without creating a separate process ?
(I assume it would be mainly a question of code separation between the "tool" exposition, and the underlying code, but I might be wrong about this...)

Thanks,

**Comment by sensslen** [comment]
I can answer this for the 3.0.0 alpha branch. There it would be very easy, as the architecture of the application simply uses the Program.cs to bind things together. Pull Reuqests moving towards this direction are very welcome.

**Comment by Greybird** [comment]
Thanks,
I'll see if I can make a proposal on the 3.0.0 branch to help!
